data/misc/gummi.1
.TH GUMMI 1 "APRIL 2011" Linux "User Manuals"
.SH NAME
gummi \- LaTeX editor
.SH DESCRIPTION
.B gummi
is a simple LaTeX editor for GTK+.
.SH OPTIONS
.TP
\-d
run gummi with debug logging enabled.
.TP
\-h
display command line options.
.SH BUGS
Undoubtedly, please report at:
https://github.com/alexandervdm/gummi
.SH AUTHORS
Alexander van der Meij <gummi{pleasenospam}vandermeij.tech>
.PP
Wei-Ning Huang <aitjcize{pleasenospam}gmail.com>


ChangeLog
-----------------------------------------------------------------------------
Gummi CHANGELOG
-----------------------------------------------------------------------------

Gummi 0.6.6 "We're still here" (January 27th, 2016):

	Bugs fixed:
	* Fix using predictable filenames in /tmp (CVE-2015-7758) 
	* Fix source id warnings during motion runtime
	* Fix open-project crash due to missing file chooser title
    * Fix synctex for input files with relative paths
	* Add descriptive keywords to our desktop entry file
	* Improvements to the bibliography detection regex
	* Only allow loading tex files through file->open
    * Desensitize cut/copy tools when no text is selected
	* Various improvements to build system
	* Removed deprecated code

Gummi 0.6.5 (October 18th, 2012):

	Bugs fixed:
	* Fixed empty build logs when special characters were included
	* Fixed a rare deadlock situation on program exit
	* Fixed regression in spell-checking language listing
	* Fixed regression in killing inresponsive compilation processes
	* Correctly close file descriptors for child processes
	* Implemented --version CLI argument
	* Several translation updates
	* Added Hungarian translation

Gummi 0.6.4 (June 24th, 2012):

	Bugs fixed:
	* Replaced faulty importer implementation with a toolbar variant
	* Fixed an issue where the previewer would not display correctly
	* dvi script uses exit status to signal compilation status
	* Improved the reported messages for some errors. 
	* Better TeXLive version detection
	* Fixed a reference problem that could lead to a crash
	* Search window is correctly centered in GNOME 3
	* Tab markup not enabled when project loading fails
	* Minor UI tweaks to the project & bibliography panels
	* Minor UI tweaks to the snippet configuration interface

Gummi 0.6.3 (March 28th, 2012):
	
	Bugs fixed:
	* Fix bibtex regression since 0.6.2
	* Fix for compiling through dvi/ps
	* Correctly disable template open button when none exist
	* Gummi toolbar can be hidden again
	* Fix LaTeX error screen and error line output
	* Menu item compatibility with Ubuntu's Unity fixed
	* Added Chinese (simplified) translation

Gummi 0.6.2 (March 18th, 2012):

	Bugs fixed:
	* Recovery mode is also triggered when gummi is run with file argv
	* Incomplete compilation processes are killed on tab switch
	* Fix template addition crash when zero tabs are opened
	* PDF files can now be used by the image importer
	* Fixed building/compilation scripts for explicit linking
	* Default text only written to disk when changes are made
	* SyncTeX auto-scrolling is now disabled by default
	* Converted snippet offsets to utf8
	* Bibliography items with all capital letters can be parsed
	* Various minor spelling/phrasing errors
	* Build files can now be removed from the menu
	* Lots of Windows specific issues
	* Addition of user guide link

Gummi 0.6.1 (December 2th, 2011):

	Bugs fixed:
	* Selecting text no longer freezes during updates
	* Shell escape option is now set/unset correctly
	* Fixed crash when trying to open non-existing file from cli
	* Popup windows no longer block gui in fullscreen mode
	* Documents containing input commands can pass precompile check
	* Opening snippet menu no longer blocks gui
	* Several fixes to snippet insertion mechanism
	* Minor tweaks to the build scripts for convenience
	* Fixed some minor memory leaks

Gummi 0.6.0 (November 14th, 2011):

	Bugs fixed:
	* Improved preview scaling performance
	* Importer file choosers now default to CWD
	* Fixed image import warning
	* Fixed rare segfault on gummi start
	* Search results colour depends on color theme
	* Preview scroll position restored after error
	* Replaced all win32 incompatible code
	* Configuration I/O transaction fixes
	* Modifier keys no longer trigger auto-compile
	* Correct encoded text is written on file save
	* Fixed a rare GUI hang when program is closed
	* Several GUI and icon related problems
	* Several memory leaks

	Features added:
	* Multi-tab/document support
	* LaTeX build systems (rubber/latexmk)
	* Compiling through DVI/Postscript
	* Continuous PDF preview
	* Filter function for bibliographies
	* Improved document statistics
	* Infoscreen panel with document status
	* More configurable options
	* Minimal project support
	* SyncTeX support
	* New & updated translations

Gummi 0.5.8 (March 17th, 2011):

	Bugs fixed:
	* PDF latency when document is scrolled fixed
	* Fixed gui hanging issues in certain situations
	* Fixed compatibility with Bibtex 0.99d (texlive 2010)
	* Updated glade files for compability with new releases
	* Better detection for damaged or incomplete pdf files
	* Replaced textit with emph for italic toolbar function
	* Solved various pdf rendering problems
	* High resolution gummi icon

	Features added:
	* LaTeX compilation is now a threaded process
	* Implemented Snippets interface
	* Automatic export of the PDF file on document save
	* Added German, Spanish, Polish and Czech translations
	* Updated several other translation files

Gummi 0.5.5 (December 10th, 2010):

	Bugs fixed:
	* Fixed critical search & replace regression.. :(

Gummi 0.5.4 (December 8th, 2010):

	Bugs fixed:
	* Fixed segfault on document statistics for filepaths with spaces
	* Fixed segfault on accessing non-existing recent file
	* Fixed a rare segfault that occured on opening file
	* Set default search & replace options to clone gedit's
	* Fixed cursor repeatedly jumping to error lines
	* Fixed statusbar message when a document is loaded from template
	* Multiple lines will now be tagged when they contain errors
	* Fixed PDF preview delay
	* Misc small bugfixes here and there
	* Updated translations

	Features added:
	* Added editor style scheme support
	* Added manual preview compile function
	* Document recovery mode in case of a crash
	* Allow custom compilation scripts and flags
	* Added polish & spanish translations
	
Gummi 0.5.3 (October 24th, 2010):

    Bugs fixed:
    * Reduce number of I/O calls for configuration

src/gummi.c
/**
 * @file   gummi.c
 * @brief
 *
 * Copyright (C) 2010 Gummi-Dev Team <alexvandermey@gmail.com>
 * All Rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person
 * obtaining a copy of this software and associated documentation
 * files (the "Software"), to deal in the Software without
 * restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following
 * conditions:
 *
 * The above copyright notice and this permission notice shall be
 * included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
 * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
 * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 * OTHER DEALINGS IN THE SOFTWARE.
 */

#include <glib.h>
#include <glib/gstdio.h>
#include <gtk/gtk.h>
#include <locale.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "configfile.h"
#include "environment.h"
#include "gummi-gui.h"
#include "importer.h"
#include "iofunctions.h"
#include "template.h"
#include "utils.h"
#include "biblio.h"

static int debug = 0;
Gummi* gummi = 0;

static GOptionEntry entries[] = {
    { (const gchar*)"debug", (gchar)'d', 0, G_OPTION_ARG_NONE, &debug, 
        (gchar*)"show debug info", NULL},
    { NULL, 0, 0, G_OPTION_ARG_NONE, NULL, NULL, NULL }
};

void on_window_destroy (GtkObject *object, gpointer user_data) {
    gtk_main_quit();
}

int main (int argc, char *argv[]) {
    gchar* configname;
    GtkBuilder* builder;
    GummiGui* gui;
    GuFileInfo* finfo;
    GuEditor* editor;
    GuImporter* importer;
    GuMotion* motion;
    GuPreview* preview;
    GuTemplate* templ;
    GuBiblio* biblio;

    /* set up i18n */
    bindtextdomain(PACKAGE, LOCALEDIR);
    setlocale(LC_ALL, "");
    textdomain(PACKAGE);

    GError* error = NULL;
    GOptionContext* context = g_option_context_new("files");
    g_option_context_add_main_entries(context, entries, PACKAGE);
    g_option_context_parse(context, &argc, &argv, &error);

    slog_init(debug);
    slog(L_INFO, PACKAGE_NAME" version: "PACKAGE_VERSION"\n");

    /* set up configuration file */
    configname = g_build_filename(g_get_user_config_dir(), "gummi",
                                  "gummi.cfg", NULL);
    config_init(configname);
    slog(L_INFO, "configuration file: %s\n", configname);
    g_free(configname);

    /* initialize gtk */
    gtk_init (&argc, &argv);
    builder = gtk_builder_new();
    gtk_builder_add_from_file(builder, DATADIR"/gummi.glade", NULL);
    gtk_builder_set_translation_domain(builder, PACKAGE);

    /* initialize classes */
    gui = gui_init(builder);
    finfo = fileinfo_init();
    editor = editor_init(builder, finfo);
    importer = importer_init(builder, finfo);
    preview = preview_init(builder, finfo);
    motion = motion_init(builder, finfo, editor, preview); 
    biblio = biblio_init(builder, finfo);
    templ = template_init(builder, finfo);

    gummi = gummi_init(gui, finfo, editor, importer, motion, preview, biblio,
            templ);

    slog_set_gui_parent(gui->mainwindow);

    if (argc != 2) {
        iofunctions_load_default_text(editor);
        gummi_create_environment(gummi, NULL);
    } else {
        iofunctions_load_file(editor, argv[1]);
        gummi_create_environment(gummi, argv[1]);
    }

    gui_main(builder);
    return 0;
}
